pkg/pod/pod.go

```diff
@@ -322,6 +327,36 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		return nil, err
 	}
 
+	readonly := true
+	if config.IsSpireEnabled(ctx) {
+		volumes = append(volumes, corev1.Volume{
+			Name: spire.WorkloadAPI,
+			VolumeSource: corev1.VolumeSource{
+				CSI: &corev1.CSIVolumeSource{
+					Driver:   "csi.spiffe.io",
+					ReadOnly: &readonly,
+				},
+			},
+		})
+
+		for i := range stepContainers {
+			c := &stepContainers[i]
+			c.VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+				Name:      spire.WorkloadAPI,
+				MountPath: spire.VolumeMountPath,
+				ReadOnly:  true,
+			})
+		}
+		for i := range initContainers {
+			c := &initContainers[i]
+			c.VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+				Name:      spire.WorkloadAPI,
+				MountPath: spire.VolumeMountPath,
+				ReadOnly:  true,
+			})
+		}
+	}
```

Comment:
Question: If users define sidecars, do we also need to mount these volumes there? Can users even produce results from sidecars (probably not so we may not want to worry)? Eventually, when we enable spire for sidecar logs, we will have to mount this volume in that specific sidecar.

Reply:
Thats a good call out. Sidecars are given access to /tekton/results by the volume mounts.
Currently we are not mounting the SPIRE volumes into user defined sidecars as that would mean that the sidecars will have to initiate the signing, which might not work.
But for sidecar logs we will need to mount these volumes.

Reply:
Please add this consideration to the issue I mentioned above for SPIRE + larger Results via Sidecar logs 🙏🏾

Reply:
Updated this in the issue.